
# flight mission
STATIC_GATE_TEST: false
mission_period: 8.0 #s

drone:
  wing_len: 0.8
  height: 0.3

# gate parameters  
gate:
  length: 1.2
  width: 0.56
  linear_vel: [0.0, 0.0, 0.0]
  angular_vel: 0.0 
  linear_vel: [-0.5,-0.3,0.0]
  angular_vel: -1.57
traj_server:
  takeoff_height: 1.2
  drone_yaw: 0.0 # take-off yaw
mission:
  #################################################
  ######--------- for test --------#########
  #################################################
  # ##------ MPC initial pose ----##
  ## ##initial pose only works for python sim
  # initial_position: [0.0, -3.0, 1]
  # initial_ori_euler: [0.0, 0.0, 0.0] # RPY, only yaw is active

  # ##------ gate pose ----##
  # # only feasible for STATIC_GATE_TEST
  # gate_position: [1.0, 0.0, 1.5]
  # gate_ori_RP: [-0.0, -1.055, 0.0]

  # ##------ goal pose ----##
  # goal_position: [0.0, 3, 1]
  # goal_ori_euler: [0.0, 0.0, 0.0] # RPY, only yaw is active

  #################################################
  ######-----end of for testing --------------#####
  #################################################
  #################################################
  ######------------- Training -----------#########
  #################################################
  ##------ MPC initial pose ----##
  ##initial pose only works for python sim
  initial_position: [0.0, -6, 0]
  initial_ori_euler: [0.0, 0.0, 0] # RPY, only yaw is active

  ##------ gate pose ----##
  # only feasible for STATIC_GATE_TEST
  gate_position: [1.0, -0.0, 1.5]
  gate_ori_RP: [-0.0, -0.99, -0.0]  # 1 for 90 degree

  ##------ goal pose ----##
  goal_position: [0.0, 3, 0]
  goal_ori_euler: [0.0, 0.0,0.0] # RPY, only yaw is active

  
# MPC parameters
learning_agile:

  ##------ fix weight parameters ----##
  # wrt: ,gate traverse position cost
  # wqt: gate traverse attitude cost
  # wthrust: input thrust cost
  # wwt: input angular velocity cost
  # wrf: final position cost
  # wvf: final velocity cost
  # wqf: final attitude cost
  # wwf: final angular velocity cost
  
  ##-----parameters for testing ------##
  # wrt: 5.0
  # wqt: 80.0
  # wthrust: 1.0
  # wwt: 10.0
  # wrf: 5.0
  # wvf: 1.0
  # wqf: 1.0

  ##-----parameters for training ------## 
  # #### for a stabler training,######## 
  wrt: 5.0
  wqt: 10.0
  wthrust: 1.0
  wwt: 5.0
  wwt_z: 5
  
  # path cost
  wrp: 10        ## too small, large deviation to the goal
  wvp: 0.01
  wqp: 0.01

  # terminal cost
  wrf: 500.0     ## large, unstable
  wvf: 100.0
  wqf: 100.0
  ##------ traversing parameters ---##
  max_traverse_weight: 100 #100
  traverse_weight_span: 10
  traverse_time: 1.5 # only for STATIC_GATE_TEST

  ##------ safety parameters  ------##
  no_solution_flag_t_thresh: 0.1
  
  ##------ MPC problem parameters---## 
  ## ---- hard coded in the solver ----##
  ## modify this required recompilation of the solver
  horizon: 40
  dt: 0.1
    
  ##------ model parameters --------##
  ## ---- hard coded in the solver ----##
  ## modify this required recompilation of the solver
  single_motor_max_thrust: 2.0 #2.1334185 #N
  # 1.7268 N is based on the hover throttle: 0.29
  # drone mass: 0.248kg, g: 9.81m/s^2 weight: 0.248*9.81 = 2.43368N
  # hover throttle: 0.29, thrust: 0.29*max_thrust =2.43368N
  # max_thrust = 2.43368/0.29 = 8.4N
  # single_motor_max_thrust = 8.4/4 = 2.1N

  
  ##------ constraints parameters --##
  ## ---- hard coded in the solver ----##
  ## modify this required recompilation of the solver

  # throttle constraints
  throttle_upper_bound: 0.8
  throttle_lower_bound: 0.2


  # position constraints
  pos_bound: 15 #m (-b,b)

  # velocity constraints
  linear_vel_bound: 15 #m/s

  # angular velocity constraints
  angular_vel_bound_xy: 15.57 #rad/s
  angular_vel_bound_z: 15.57 #rad/s


  ## ----- miscellaneous parameters  ----##
  pred_traj_vis: true


  ##------  training parameters  ------##
training_param:
  desired_average_vel: 1.0


## ------NN configuration ----##
NN2_model_name: 'NN2_imitate_1.pth'
NN2_freq: 10